js/components/closed-caption/colorSelectionTab.js

```diff
@@ -1,9 +1,91 @@
-var React = require('react');
+var React = require('react'),
+    SelectionContainer = require('./selectionContainer'),
+    ColorSelector = require('../colorSelector');
 
 var ColorSelectionTab = React.createClass({
+
+  getInitialState: function() {
+    return {
+      selectedTextColor: this.props.closedCaptionOptions.textColor,
+      selectedWindowColor: this.props.closedCaptionOptions.windowColor,
+      selectedBackgroundColor: this.props.closedCaptionOptions.backgroundColor
+    };
+  },
+
+  changeTextColor: function(color){
+    if (this.props.closedCaptionOptions.enabled){
+      this.props.controller.onClosedCaptionTextColorChange(color);
+      this.setState({
+        selectedTextColor: color
+      });
+    }
+  },
+
+  changeWindowColor: function(color){
+    if (this.props.closedCaptionOptions.enabled){
+      this.props.controller.onClosedCaptionWindowColorChange(color);
+      this.setState({
+        selectedWindowColor: color
+      });
+    }
+  },
+
+  changeBackgroundColor: function(color){
+    if (this.props.closedCaptionOptions.enabled){
+      this.props.controller.onClosedCaptionBackgroundColorChange(color);
+      this.setState({
+        selectedBackgroundColor: color
+      });
+    }
+  },
+
   render: function(){
+    var textColors = ["White", "Blue", "Magenta", "Green", "Yellow", "Red", "Cyan", "Black"];
+    var windowColors = ["Transparent", "White", "Blue", "Magenta", "Green", "Yellow", "Red", "Cyan", "Black"];
+    var backgroundColors = ["Transparent", "White", "Blue", "Magenta", "Green", "Yellow", "Red", "Cyan", "Black"];
```

Comment:
These can be moved to the getInitialState method and assigned to the "this" object to avoid declaring and allocating these arrays on every render call.